bin/inference/pycbc_inference_plot_posterior

```diff
@@ -26,92 +26,167 @@
+
+# get likelihood statistic values
 if opts.z_arg is not None:
     logging.info("Getting likelihood stats")
-    likelihood_stats = fp.read_likelihood_stats(thin_start=opts.thin_start,
-            thin_end=opts.thin_end, thin_interval=opts.thin_interval,
-            iteration=opts.iteration)
-    zvals, zlbl = option_utils.get_zvalues(fp, opts.z_arg, likelihood_stats)
-    show_colorbar = True
+
+    # lists to hold z-axis values and labels for each input file
+    zvals = []
+    zlbl = []
+
+    # loop over each input file and append z-axis values and labels to lists
+    for f in fp:
+        likelihood_stats = f.read_likelihood_stats(
+            thin_start=opts.thin_start, thin_end=opts.thin_end,
+            thin_interval=opts.thin_interval, iteration=opts.iteration)
+        f_zvals, f_zlbl = option_utils.get_zvalues(f, opts.z_arg,
+                                                   likelihood_stats)
+        zvals.append(f_zvals)
+        zlbl.append(f_zlbl)
+        f.close()
+
+# else there are no z-axis values
 else:
     zvals = None
     zlbl = None
-    show_colorbar = False
-fp.close()
+    for f in fp:
+        f.close()
 
-# If no plotting options selected, set default options based on number
-# of parameters.
+# determine if colorbar should be shown
+show_colorbar = True if opts.z_arg else False
+
+# if no plotting options selected, then the default options are based
+# on the number of parameters
 plot_options = [opts.plot_marginal, opts.plot_scatter, opts.plot_density]
 if not numpy.any(plot_options):
-    if len(parameters) == 1:
+    if len(parameters[0]) == 1:
         opts.plot_marginal = True
     else:
         opts.plot_scatter = True
-        # FIXME: right now if there are two parameters it wants both plot_scatter
-        # and plot_marginal. One should have the option of give only plot_scatter
-        # and that should be the default for two or more parameters
+        # FIXME: right now if there are two parameters it wants
+        # both plot_scatter and plot_marginal. One should have the option
+        # of give only plot_scatter and that should be the default for
+        # two or more parameters
         opts.plot_marginal = True
 
+# get minimum and maximum ranges for each parameter from command line
 mins, maxs = option_utils.plot_ranges_from_cli(opts)
+
 # add any missing parameters
-for p in parameters:
+for p in parameters[0]:
     if p not in mins:
-        mins[p] = samples[p].min()
-for p in parameters:
+        mins[p] = numpy.array([s[p].min() for s in samples]).min()
     if p not in maxs:
-        maxs[p] = samples[p].max()
+        maxs[p] = numpy.array([s[p].max() for s in samples]).max()
 
+# get expected parameter values from command line
 expected_parameters = option_utils.expected_parameters_from_cli(opts)
 
+# assign some colors
+colors = itertools.cycle(["black", "red", "cyan", "magenta",
+                          "green", "blue", "yellow"])
```

Comment:
Use [matplotlib's color cycle](https://matplotlib.org/users/dflt_style_changes.html#colors-in-default-property-cycle) instead. Namely, just do `'C{}'.format(i)` in the cycle below.

Reply:
> in the cycle below

You mean the cycle above? Also that color map puts green and red next to each other at the beginning, I was trying to avoid that.

Reply:
Sorry, above. That's a good point about putting red and green next to each other. I'm surprised that mpl didn't address that in the updated color cycle. Still, I like the new color cycle that mpl has. How about move C3 (red) to after C7 (gray)? Also keep the first color black. Alternatively, you could try using a color map, such as inferno. Then it'd also be black-and-white printer friendly. Not sure how good that'd look though.

I'm not going to hold up approval of this because of the color scheme, so if you prefer to keep it as-is, do so.